# **Health Supply chain analytics project: Data preparation**

To get clear insights, The data must be analyzed, key variables identified, and the dataset ensured it is ready for analysis.

##   Step 1: Build the dataframe
Create a pandas dataframe for data learning, and future exploratory data analysis (EDA) and statistical activities

In [13]:
import pandas as pd               #library exercise for buidling dataframes
import numpy as np                #numpy is imported with pandas
#Load data from the CSV extracted from the database
df1=pd.read_csv("data/requisition1.csv")
df2=pd.read_csv("data/requisition2.csv")
df3=pd.read_csv("data/requisition3.csv")
df4=pd.read_csv("data/requisition4.csv")
df5=pd.read_csv("data/requisition5.csv")
df6=pd.read_csv("data/requisition6.csv")

In [40]:
# Data translation (To skip or ignor) during the post of scripts
fr_to_en={'Août':'Aug'}
for fr,en in fr_to_en.items():
    df1['processing_periods_name']=df1['processing_periods_name'].str.replace(fr,en,regex=False)
    df2['processing_periods_name']=df2['processing_periods_name'].str.replace(fr,en,regex=False)
    df3['processing_periods_name']=df3['processing_periods_name'].str.replace(fr,en,regex=False)
    df4['processing_periods_name']=df4['processing_periods_name'].str.replace(fr,en,regex=False)
    df5['processing_periods_name']=df5['processing_periods_name'].str.replace(fr,en,regex=False)
    df6['processing_periods_name']=df6['processing_periods_name'].str.replace(fr,en,regex=False)

In [126]:
df1.to_csv("data/requisition1.csv",index=False)
df2.to_csv("data/requisition2.csv",index=False)
df3.to_csv("data/requisition3.csv",index=False)
df4.to_csv("data/requisition4.csv",index=False)
df5.to_csv("data/requisition5.csv",index=False)
df6.to_csv("data/requisition6.csv",index=False)

##   Step 2: Understand the data - Inspect the data

In [91]:
df1.head()

,product_primaryname,processing_periods_name,facility_type_name,beginningbalance,quantityreceived,quantitydispensed,stockinhand,quantityrequested,quantityapproved,totallossesandadjustments,amc,packsize,zone,dispensingunit
0,PYRA-COMP,May 2025,HEALTH CENTER,290.0,22.0,10.0,980.0,NaN,NaN,313,2.0,25,DISTRICT25,TABLET
1,PYRA-COMP,May 2025,HEALTH CENTER,3510.0,6.0,437.0,1426.0,NaN,0.0,246,151.0,30,DISTRICT28,TABLET
2,ACT-GC,Nov 2025,HEALTH CENTER,125.0,NaN,52.0,140.0,1433.0,NaN,25,NaN,1000,DISTRICT1,BLISTER
3,PYRA-SAC,Dec 2024,HEALTH CENTER,1015.0,209.0,NaN,533.0,810.0,327.0,408,NaN,25,DISTRICT1,SACHET
4,SP,Jan 2024,HEALTH CENTER,2805.0,NaN,528.0,NaN,NaN,730.0,42,448.0,30,DISTRICT13,TABLET


### Append the dataframes. Check the structure of all the copies of the df. The number of the #columns should match before appending them

In [92]:
# Inspect the status of one of the dataframe
df1.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 28446 entries, 0 to 28445
Data columns (total 14 columns):
 #   Column                     Non-Null Count  Dtype  
---  ------                     --------------  -----  
 0   product_primaryname        28446 non-null  object 
 1   processing_periods_name    28446 non-null  object 
 2   facility_type_name         28446 non-null  object 
 3   beginningbalance           20273 non-null  float64
 4   quantityreceived           20413 non-null  float64
 5   quantitydispensed          20274 non-null  float64
 6   stockinhand                20313 non-null  float64
 7   quantityrequested          7856 non-null   float64
 8   quantityapproved           23020 non-null  float64
 9   totallossesandadjustments  28446 non-null  int64  
 10  amc                        20393 non-null  float64
 11  packsize                   28446 non-null  int64  
 12  zone                       28446 non-null  object 
 13  dispensingunit             28446 non-null  obj

In [93]:
print("df1 columns match df2: ",df1.columns.equals(df2.columns))
print("df1 columns match df3: ",df1.columns.equals(df3.columns))
print("df1 columns match df4: ",df1.columns.equals(df4.columns))
print("df1 columns match df5: ",df1.columns.equals(df5.columns))
print("df1 columns match df6: ",df1.columns.equals(df6.columns))


df1 columns match df2:  True
df1 columns match df3:  True
df1 columns match df4:  True
df1 columns match df5:  True
df1 columns match df6:  True


In [94]:
requisitions_all=pd.concat([df1,df2,df3,df4,df5,df6])

In [95]:
requisitions_all.info()

<class 'pandas.core.frame.DataFrame'>
Index: 170676 entries, 0 to 28445
Data columns (total 14 columns):
 #   Column                     Non-Null Count   Dtype  
---  ------                     --------------   -----  
 0   product_primaryname        170676 non-null  object 
 1   processing_periods_name    170676 non-null  object 
 2   facility_type_name         170676 non-null  object 
 3   beginningbalance           121720 non-null  float64
 4   quantityreceived           121980 non-null  float64
 5   quantitydispensed          121833 non-null  float64
 6   stockinhand                121764 non-null  float64
 7   quantityrequested          47072 non-null   float64
 8   quantityapproved           138687 non-null  float64
 9   totallossesandadjustments  170676 non-null  int64  
 10  amc                        122104 non-null  float64
 11  packsize                   170676 non-null  int64  
 12  zone                       170676 non-null  object 
 13  dispensingunit             170676 n

#### Delete IDs
The req_id,req_line_id,zone_id,The processing_periods_id are ID and do not represent any meaningfull values. We can remove them

In [69]:
#requisitions_all.drop(columns=['product_code','processing_periods_id','facility_code','remarks',
#                              'skipped','req_id','req_line_id','req_status','zone_id'],
#                      inplace=True)

In [96]:
#Check if the merge of dataframe happens successfully by comparing the number of rows with requisitions_all
print("Total number of merged row: ",df1.shape[0]+df2.shape[0]+df3.shape[0]+df4.shape[0]+df5.shape[0]+df6.shape[0])

Total number of merged row:  170676


In [71]:
#requisitions_all=synthetic_data

In [97]:
requisitions_all.head(20)

,product_primaryname,processing_periods_name,facility_type_name,beginningbalance,quantityreceived,quantitydispensed,stockinhand,quantityrequested,quantityapproved,totallossesandadjustments,amc,packsize,zone,dispensingunit
0,PYRA-COMP,May 2025,HEALTH CENTER,290.0,22.0,10.0,980.0,NaN,NaN,313,2.0,25,DISTRICT25,TABLET
1,PYRA-COMP,May 2025,HEALTH CENTER,3510.0,6.0,437.0,1426.0,NaN,0.0,246,151.0,30,DISTRICT28,TABLET
2,ACT-GC,Nov 2025,HEALTH CENTER,125.0,NaN,52.0,140.0,1433.0,NaN,25,NaN,1000,DISTRICT1,BLISTER
3,PYRA-SAC,Dec 2024,HEALTH CENTER,1015.0,209.0,NaN,533.0,810.0,327.0,408,NaN,25,DISTRICT1,SACHET
4,SP,Jan 2024,HEALTH CENTER,2805.0,NaN,528.0,NaN,NaN,730.0,42,448.0,30,DISTRICT13,TABLET
5,ART-INJ,Jul 2025,HEALTH CENTER,1744.0,4.0,409.0,1508.0,NaN,7396.0,-978,681.0,1000,DISTRICT28,VIAL
6,RDT,Apr 2025,HEALTH CENTER,3187.0,737.0,471.0,3934.0,NaN,NaN,345,340.0,30,DISTRICT2,PIECE
7,ITN,Jan 2024,HEALTH CENTER,NaN,NaN,452.0,743.0,NaN,3699.0,-769,121.0,30,DISTRICT28,PIECE
8,ART-INJ,Dec 2023,HEALTH CENTER,NaN,0.0,NaN,NaN,NaN,NaN,133,0.0,1,DISTRICT8,VIAL
9,ACT-INF,Jan 2024,DISTRICT HOSPITAL,NaN,NaN,NaN,NaN,614.0,0.0,331,10.0,30,DISTRICT24,BLISTER


## Step 2: Understand the data - Investigate vardiable the data

In [98]:
requisitions_all.info()

<class 'pandas.core.frame.DataFrame'>
Index: 170676 entries, 0 to 28445
Data columns (total 14 columns):
 #   Column                     Non-Null Count   Dtype  
---  ------                     --------------   -----  
 0   product_primaryname        170676 non-null  object 
 1   processing_periods_name    170676 non-null  object 
 2   facility_type_name         170676 non-null  object 
 3   beginningbalance           121720 non-null  float64
 4   quantityreceived           121980 non-null  float64
 5   quantitydispensed          121833 non-null  float64
 6   stockinhand                121764 non-null  float64
 7   quantityrequested          47072 non-null   float64
 8   quantityapproved           138687 non-null  float64
 9   totallossesandadjustments  170676 non-null  int64  
 10  amc                        122104 non-null  float64
 11  packsize                   170676 non-null  int64  
 12  zone                       170676 non-null  object 
 13  dispensingunit             170676 n

In [99]:
requisitions_all.describe()

,beginningbalance,quantityreceived,quantitydispensed,stockinhand,quantityrequested,quantityapproved,totallossesandadjustments,amc,packsize
count,121720.000000,121980.000000,121833.000000,121764.000000,47072.000000,138687.000000,170676.000000,122104.000000,170676.000000
mean,1158.502358,201.675422,301.538598,1123.561553,758.568533,818.731410,-5.851836,248.920903,185.914499
std,1418.286005,467.469287,346.968959,1346.303309,495.714502,2019.531493,430.988308,345.346524,365.019260
min,0.000000,0.000000,0.000000,1.000000,7.000000,0.000000,-1903.000000,0.000000,1.000000
25%,197.000000,0.000000,62.000000,200.000000,396.000000,1.000000,-296.000000,28.000000,2.000000
50%,651.500000,17.000000,183.000000,649.000000,652.000000,42.000000,-4.000000,117.000000,30.000000
75%,1593.000000,170.000000,417.000000,1549.000000,1008.000000,594.000000,286.000000,333.000000,50.000000
max,19998.000000,9766.000000,5077.000000,16988.000000,4700.000000,52652.000000,2091.000000,4862.000000,1000.000000


### 2.1. Understand Numerical data

* The quantityreceived means that at least 25% of the observations have no receipt. The median is low (17) which means that most receipts are small or occurance within a year is low. This is normal as the reception is supposed to happens on quarterly basis. Long tail (9766) could also refers occasional large deliveries which can be due to the dispensing units or packsize.
* The quantitydispensed  with no zero quartile means that facilities consistently dispense.Strong spread heterogeneous facility demand expresed by widely dispersed values.Different facilities have very different consumption levels. Long tail high-burden facilities or peak seasons due to the fact that some facilities consistently dispense much more or some peak saisons for high malaria transmission. The max value of 16,988 show a potential inefficiency such as overstock, campaign stock or a problem related to the dispensing units.
* Stockinhand shows indicates that facilities generally maintain positive stock levels, with no evidence of widespread stockouts within the year.The wider variability (1549-200) shows that facilities hold very diffrent stock level. Very similar medians beginningbalance ≈ 651 and stockinhand median ≈ 649 prove that data consistency looks good.
* Quantityrequested count of 47,072 is much lower than others. This means that there are many missing values due to the fact that orders are placed only in the order months.  means that at least 25% of the observations are equal to 0. This could be related to the 25% of the facilities that do not place any order. the tigh variability shows that order behavior is structured.
* Quantityapproved is oftem lower then requested median=42 vs 652,show that facilities request much more than what they received. max approved = 52,652 vs max approved = 52,652 means that 
* totallossesandadjustments with a slight negative  (centered near 0) means  more negative adjustments or losses.
* AMC shows smooth distribution with no zeros at the quartiles but with high variability 28 to 4,862.
* packsize varies from 1 to 1000 means that the products are ranging from single dispensing units (e.g., individual tablets or tests) to bulk packages containing up to 1000 tablets (product heterogeneity) . All quantities are represented based on their packsize.

#### 2.2. Dealing with missing values

In [23]:
requisitions_all.shape[0]

170676

In [100]:
requisitions_all.isna().sum()

product_primaryname               0
processing_periods_name           0
facility_type_name                0
beginningbalance              48956
quantityreceived              48696
quantitydispensed             48843
stockinhand                   48912
quantityrequested            123604
quantityapproved              31989
totallossesandadjustments         0
amc                           48572
packsize                          0
zone                              0
dispensingunit                    0
dtype: int64

Let's represent the missing values in %

In [101]:
(requisitions_all.isna().sum()/requisitions_all.shape[0])*100

product_primaryname           0.000000
processing_periods_name       0.000000
facility_type_name            0.000000
beginningbalance             28.683588
quantityreceived             28.531252
quantitydispensed            28.617380
stockinhand                  28.657808
quantityrequested            72.420258
quantityapproved             18.742530
totallossesandadjustments     0.000000
amc                          28.458600
packsize                      0.000000
zone                          0.000000
dispensingunit                0.000000
dtype: float64

* There are missing values in the following columns:beginningbalance => 48956, quantityreceived => 48696. etc. Same for quantitydispensed,stockinhand,quantityrequested, quantityapproved,quantityapproved,amc.
* The core inventory variables beginningbalance,quantityreceived,quantitydispensed,stockinhand,amc are all representing ~ 28% (48,000) observations.These variables are jointly missing, which is very important could suggest entire reports are missing (not individual fields). This may be due to the reporting completeness issue, facilities that did NOT submit reports for that month/product as the product is not available, not used.
* quantityrequested represent ~72% missing(123604), this pattern could be justified as requests only happen in order's months - quarterly (requests only happen in order months) so Feb, May, Aug, Nov with no requests and 4 months with requests.
* Quantityapproved represent ~19% missing(31989). This is lower than quantity requested, it could du to the fact that the orders are placed even for facilities with no requested quantities (pushing strategy formalized)

In [102]:
# Display the missing data 
mask=requisitions_all[
    ["beginningbalance","quantityreceived","quantitydispensed","stockinhand"]
].isna().all(axis=1)
mask_amc=requisitions_all[
    ["amc"]
].isna().all(axis=1)

In [103]:
requisitions_all[mask].head(10)

,product_primaryname,processing_periods_name,facility_type_name,beginningbalance,quantityreceived,quantitydispensed,stockinhand,quantityrequested,quantityapproved,totallossesandadjustments,amc,packsize,zone,dispensingunit
9,ACT-INF,Jan 2024,DISTRICT HOSPITAL,NaN,NaN,NaN,NaN,614.0,0.0,331,10.0,30,DISTRICT24,BLISTER
60,QUIN,Nov 2024,DISTRICT HOSPITAL,NaN,NaN,NaN,NaN,793.0,4.0,-185,NaN,30,DISTRICT7,TABLET
323,SP,Jan 2024,HEALTH CENTER,NaN,NaN,NaN,NaN,NaN,898.0,250,9.0,1000,DISTRICT38,TABLET
355,SP,May 2025,HEALTH CENTER,NaN,NaN,NaN,NaN,NaN,0.0,-228,NaN,1,DISTRICT13,TABLET
537,ART-SUP,Sep 2024,HEALTH CENTER,NaN,NaN,NaN,NaN,270.0,0.0,-10,0.0,25,DISTRICT26,CAPSULE
552,PYRA-SAC,Jul 2025,HEALTH CENTER,NaN,NaN,NaN,NaN,NaN,6288.0,-507,69.0,30,DISTRICT11,SACHET
877,ACT-GC,Oct 2025,HEALTH CENTER,NaN,NaN,NaN,NaN,NaN,0.0,-539,13.0,1000,DISTRICT30,BLISTER
1205,SP,Jan 2024,HEALTH CENTER,NaN,NaN,NaN,NaN,NaN,2696.0,-177,105.0,1000,DISTRICT33,TABLET
1501,ACT-GC,Feb 2025,HEALTH CENTER,NaN,NaN,NaN,NaN,NaN,767.0,-42,337.0,30,DISTRICT37,BLISTER
1505,ART-SUP,Jan 2025,HEALTH CENTER,NaN,NaN,NaN,NaN,NaN,4.0,-155,180.0,1,DISTRICT7,CAPSULE


In [104]:
requisitions_all[mask_amc].head(10)

,product_primaryname,processing_periods_name,facility_type_name,beginningbalance,quantityreceived,quantitydispensed,stockinhand,quantityrequested,quantityapproved,totallossesandadjustments,amc,packsize,zone,dispensingunit
2,ACT-GC,Nov 2025,HEALTH CENTER,125.0,NaN,52.0,140.0,1433.0,NaN,25,NaN,1000,DISTRICT1,BLISTER
3,PYRA-SAC,Dec 2024,HEALTH CENTER,1015.0,209.0,NaN,533.0,810.0,327.0,408,NaN,25,DISTRICT1,SACHET
14,SP,May 2025,HEALTH CENTER,293.0,2.0,363.0,39.0,NaN,6730.0,1216,NaN,1000,DISTRICT25,TABLET
17,QUIN,Mar 2025,HEALTH CENTER,NaN,513.0,253.0,NaN,1037.0,548.0,-606,NaN,1000,DISTRICT1,TABLET
20,ACT-GC,May 2024,DISTRICT HOSPITAL,2002.0,0.0,343.0,1844.0,NaN,434.0,252,NaN,1000,DISTRICT31,BLISTER
24,PYRA-COMP,Mar 2025,HEALTH CENTER,1846.0,865.0,756.0,1321.0,NaN,6.0,-5,NaN,1,DISTRICT36,TABLET
28,ACT-INF,Jun 2025,HEALTH CENTER,NaN,1.0,129.0,574.0,NaN,67.0,696,NaN,30,DISTRICT9,BLISTER
33,PYRA-COMP,Oct 2025,HEALTH CENTER,10481.0,337.0,1693.0,NaN,126.0,0.0,324,NaN,30,DISTRICT33,TABLET
39,ART-SUP,Nov 2024,HEALTH CENTER,NaN,434.0,NaN,NaN,NaN,1.0,-137,NaN,1,DISTRICT8,CAPSULE
45,ART-INJ,Dec 2023,HEALTH CENTER,1748.0,0.0,47.0,NaN,167.0,0.0,-597,NaN,30,DISTRICT31,VIAL


In [105]:
# Remove missing data
requisitions_cleaned_inventory=requisitions_all.dropna(
    subset=[
        "beginningbalance",
        "quantityreceived",
        "quantitydispensed",
        "stockinhand"
    ]
)


In [106]:
requisitions_cleaned_inventory=requisitions_cleaned_inventory.dropna(
    subset=[
        "amc"
    ]
)

In [107]:
requisitions_cleaned_inventory.head()

,product_primaryname,processing_periods_name,facility_type_name,beginningbalance,quantityreceived,quantitydispensed,stockinhand,quantityrequested,quantityapproved,totallossesandadjustments,amc,packsize,zone,dispensingunit
0,PYRA-COMP,May 2025,HEALTH CENTER,290.0,22.0,10.0,980.0,NaN,NaN,313,2.0,25,DISTRICT25,TABLET
1,PYRA-COMP,May 2025,HEALTH CENTER,3510.0,6.0,437.0,1426.0,NaN,0.0,246,151.0,30,DISTRICT28,TABLET
5,ART-INJ,Jul 2025,HEALTH CENTER,1744.0,4.0,409.0,1508.0,NaN,7396.0,-978,681.0,1000,DISTRICT28,VIAL
6,RDT,Apr 2025,HEALTH CENTER,3187.0,737.0,471.0,3934.0,NaN,NaN,345,340.0,30,DISTRICT2,PIECE
11,PYRA-COMP,Jul 2025,HEALTH CENTER,3665.0,0.0,706.0,2667.0,NaN,5338.0,-1042,332.0,1,DISTRICT23,TABLET


In [108]:
#Check point: Save the clean dataset outof missing values
requisitions_cleaned_inventory.to_csv("data/requisitions_cleaned.csv",index=False)

#### 2.2. Understand categorical variables

#### Understand products data

In [109]:
product_count=requisitions_cleaned_inventory['product_primaryname'].value_counts().reset_index()

In [110]:
product_count.columns=['Product','Occurence']
product_count.sort_values(['Occurence'],ascending=False)

,Product,Occurence
0,QUIN,2714
1,RDT,2711
2,ART-SUP,2684
3,ACT-GC,2667
4,ACT-AD,2660
5,ACT-INF,2656
6,ART-INJ,2650
7,ACT-SC,2639
8,ITN,2613
9,SP,2610


* This table is very rich analytically, even though it looks simple. It tells you a lot about the structure, completeness, and design of your supply chain dataset.There is 11 products in the dataset. All product occurance or obserations are almost equals. The highest is the QUIN=>2714 and the lowest is the PYRA-SAC=>2540, the difference 174 is very small.
* This pattern strongly suggests that each product is expected to be reported systematically

#### Understand reporting periods data

In [111]:
requisitions_cleaned_inventory['processing_periods_name'].drop_duplicates().sort_values()

91     Apr 2024
6      Apr 2025
40     Aug 2024
120    Aug 2025
25     Dec 2023
284    Dec 2024
36     Feb 2024
199    Feb 2025
37     Jan 2024
376    Jan 2025
135    Jul 2024
5      Jul 2025
163    Jun 2024
236    Jun 2025
160    Mar 2024
278    Mar 2025
73     May 2024
0      May 2025
93     Nov 2024
94     Nov 2025
175    Oct 2024
75     Oct 2025
31     Sep 2024
313    Sep 2025
Name: processing_periods_name, dtype: object

In [112]:
requisitions_cleaned_inventory['processing_periods_name'].drop_duplicates().count()

np.int64(24)

In [113]:
# Convert the content of the column 'processing_periods_name' to datetime to be able to
#handle it with operation like sorting
requisitions_cleaned_inventory['processing_periods_name']=pd.to_datetime(requisitions_cleaned_inventory['processing_periods_name'].str.strip(),
               format='%b %Y')

In [114]:
periods_count=requisitions_cleaned_inventory['processing_periods_name'].value_counts().reset_index()
periods_count.columns=['Period','Occurence']
#periods_count.sort_values(['Occurence'],ascending=True)

In [115]:
periods_count.sort_values(['Period']).reset_index()

,index,Period,Occurence
0,22,2023-12-01,1228
1,23,2024-01-01,1221
2,20,2024-02-01,1255
3,13,2024-03-01,1322
4,17,2024-04-01,1308
5,9,2024-05-01,1338
6,19,2024-06-01,1271
7,14,2024-07-01,1321
8,15,2024-08-01,1318
9,6,2024-09-01,1361


* There is 24 periods of data observations.

In [116]:
periods_count.sort_values(['Occurence'],ascending=False)

,Period,Occurence
0,2025-02-01,1388
1,2025-01-01,1386
2,2025-03-01,1383
3,2025-08-01,1370
4,2025-11-01,1370
5,2025-10-01,1367
6,2024-09-01,1361
7,2025-05-01,1359
8,2024-11-01,1343
9,2024-05-01,1338


* There is 24 periods (months) ranging from December 2023 to August 2025
* In general, we can observe increasing trend over the time. This could describe the increase of improvement of completeness over time or the intergration of new facilities in the reporting effort.

#### Understand facility_type data

In [117]:
facility_type_count=requisitions_cleaned_inventory['facility_type_name'].value_counts().reset_index()
facility_type_count.columns=['facility_type','Occurence']

In [118]:
facility_type_count

,facility_type,Occurence
0,HEALTH CENTER,29191
1,DISTRICT HOSPITAL,1929
2,REGIONAL HOSPITAL,455
3,NATIONAL HOSPITAL,167


* There is 4 facitities type
* The Health center is the type with more reports or observations, followed by the district hospital.

#### Understand Zone data

In [119]:
zone_count=requisitions_cleaned_inventory['zone'].value_counts().reset_index()
zone_count.columns=['zone','Occurence']
zone_count

,zone,Occurence
0,DISTRICT31,1679
1,DISTRICT29,1497
2,DISTRICT33,1450
3,DISTRICT28,1399
4,DISTRICT13,1368
5,DISTRICT32,1363
6,DISTRICT9,1293
7,DISTRICT23,1215
8,DISTRICT20,1212
9,DISTRICT25,1204


* There are 37 zones, some of them has more reports. This can be due to the number of facilities within each of them. 

#### Understand the dispensing_units

In [120]:
dispensingunit_count=requisitions_cleaned_inventory['dispensingunit'].value_counts().reset_index()
dispensingunit_count.columns=['dispensingunit','Occurence']
dispensingunit_count

,dispensingunit,Occurence
0,BLISTER,10622
1,TABLET,7922
2,PIECE,5324
3,CAPSULE,2684
4,VIAL,2650
5,SACHET,2540


In [122]:
#dispensingunit_countt['Occurence']/dispensingunit_count['Occurence'].sum()*100
dispensingunit_count['percentage']=dispensingunit_count['Occurence']/dispensingunit_count['Occurence'].sum()*100
dispensingunit_count

,dispensingunit,Occurence,percentage
0,BLISTER,10622,33.463550
1,TABLET,7922,24.957470
2,PIECE,5324,16.772730
3,CAPSULE,2684,8.455674
4,VIAL,2650,8.348560
5,SACHET,2540,8.002016


* There are 7 types of the dispensing units.
* The dataset is dominated by solid oral reporting. Blister and tablet represent ~ 33% of the dataset. Treatment commodities dominate the reporting volume, followed by diagnostics (Test and piece) that refers respectively to RDT and ITNs.

In [124]:
#requisitions_cleaned_inventory[['product_primaryname','dispensingunit']]
df_product_dispensingunit_unique = requisitions_cleaned_inventory[['product_primaryname', 'dispensingunit']].drop_duplicates().reset_index(drop=True)


In [125]:
df_product_dispensingunit_unique

,product_primaryname,dispensingunit
0,PYRA-COMP,TABLET
1,ART-INJ,VIAL
2,RDT,PIECE
3,ACT-GC,BLISTER
4,ACT-SC,BLISTER
5,QUIN,TABLET
6,SP,TABLET
7,ACT-AD,BLISTER
8,ART-SUP,CAPSULE
9,PYRA-SAC,SACHET
